web/src/HUD.tsx

```diff
@@ -82,10 +85,14 @@ class HUD extends Component<HudProps, HudState> {
 
   componentDidMount() {
     this.controller.createNewSocket()
+    this._isMounted = true
   }
 
   componentWillUnmount() {
-    this.controller.dispose()
+    if (this._isMounted) {
+      this.controller.dispose()
```

Comment:
sorry, i still don't understand this pr. why should the component need to be mounted to dispose the controller?